<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Differential Binding and Expression Analysis for DamID-seq Data â€¢ damidBind</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="deps/Source_Sans_Pro-0.4.10/font.css" rel="stylesheet">
<link href="deps/Source_Code_Pro-0.4.10/font.css" rel="stylesheet">
<link href="deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="deps/headroom-0.11.0/headroom.min.js"></script><script src="deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="deps/search-1.0.0/fuse.min.js"></script><script src="deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Differential Binding and Expression Analysis for DamID-seq Data">
<meta name="description" content="The damidBind package provides a straightforward formal analysis pipeline to analyse and explore differential DamID binding, gene transcription or chromatin accessibility between two conditions. The package imports processed data from DamID-seq experiments, either as external raw files in the form of binding bedGraphs and GFF/BED peak calls, or as internal lists of GRanges objects. After optionally normalising data, combining peaks across replicates and determining per-replicate peak occupancy, the package links bound loci to nearby genes. For RNA Polymerase DamID data, the package calculates occupancy over genes, and optionally calcualates the FDR of significantly-enriched gene occupancy. damidBind then uses either limma (for conventional log2 ratio DamID binding data) or NOIseq (for counts-based CATaDa chromatin accessibility data) to identify differentially-enriched regions, or differentially epxressed genes, between two conditions. The package provides a number of visualisation tools (volcano plots, Gene Ontology enrichment plots via ClusterProfiler and proportional Venn diagrams via BioVenn for downstream data exploration and analysis. An powerful, interactive IGV genome browser interface (powered by Shiny and igvShiny) allows users to rapidly and intuitively assess significant differentially-bound regions in their genomic context.">
<meta property="og:description" content="The damidBind package provides a straightforward formal analysis pipeline to analyse and explore differential DamID binding, gene transcription or chromatin accessibility between two conditions. The package imports processed data from DamID-seq experiments, either as external raw files in the form of binding bedGraphs and GFF/BED peak calls, or as internal lists of GRanges objects. After optionally normalising data, combining peaks across replicates and determining per-replicate peak occupancy, the package links bound loci to nearby genes. For RNA Polymerase DamID data, the package calculates occupancy over genes, and optionally calcualates the FDR of significantly-enriched gene occupancy. damidBind then uses either limma (for conventional log2 ratio DamID binding data) or NOIseq (for counts-based CATaDa chromatin accessibility data) to identify differentially-enriched regions, or differentially epxressed genes, between two conditions. The package provides a number of visualisation tools (volcano plots, Gene Ontology enrichment plots via ClusterProfiler and proportional Venn diagrams via BioVenn for downstream data exploration and analysis. An powerful, interactive IGV genome browser interface (powered by Shiny and igvShiny) allows users to rapidly and intuitively assess significant differentially-bound regions in their genomic context.">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="index.html">damidBind</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.10</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="articles/damidBind_vignette.html">damidBind</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/marshall-lab/damidBind/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h1></div>
<p>DamID is a highly-sensitive means to profile the genome-wide association of proteins with chromatin in living eukaryotic cells, without fixation or the use of antibodies. Cell-type specific techniques such as Targeted DamID to profile protein binding, and CATaDA to profile chromatin accessibility, have made the technique an extremely powerful tool to understand the binding of transcription factors, chromatin proteins, RNA polymerase and chromatin changes during development and disease.</p>
<p>The <code>damidBind</code> package provides a simple formal analysis pipeline to analyse and explore differential DamID binding, gene transcription or chromatin accessibility between two conditions. The package imports processed data from DamID-seq experiments, in the form of binding bedgraphs and GFF peak calls. After optionally normalising data, combining peaks across replicates and determining per-replicate peak occupancy, the package links bound loci to nearby genes. It then uses either <code>limma</code> (for conventional log2 ratio DamID binding data) or <code>NOIseq</code> (for counts-based CATaDa chromatin accessibility data) to identify differentially-enriched regions between two conditions.</p>
<p>The package provides a number of visualisation tools (volcano plots, gene ontology enrichment plots via <code>clusterProfiler</code> and proportional Venn diagrams via <code>BioVenn</code> for downstream data exploration and analysis. An powerful, interactive IGV genome browser interface (powered by <code>Shiny</code> and <code>igvShiny</code>) allows users to rapidly and intuitively assess significant differentially-bound regions in their genomic context.</p>
<p>Although extensive customisation options are available if required, much of the data handling by <code>damidBind</code> is taken care of automatically, with sensible defaults assumed. To move from loading raw data to visualising differentially-enriched regions on a volcano plot or browsing enriched regions in an interactive IGV window is a simple three command procedure.</p>
</div>
<div class="section level1">
<h1 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h1>
<p>You can install damidBind either from Bioconductor (stable) or Github (latest)</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.github.io/BiocManager/" class="external-link">"BiocManager"</a></span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># To install from Bioconductor (requires Bioconductor release 3.23):</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span>version<span class="op">=</span><span class="st">'devel'</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"damidBind"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># ... or to install the latest build from Github </span></span>
<span><span class="co"># (requires R&gt;4.4, should work on most recent Bioconductor releases):</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"marshall-lab/damidBind"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level1">
<h1 id="vignette">Vignette<a class="anchor" aria-label="anchor" href="#vignette"></a>
</h1>
<p>A more complete guide to using <code>damidBind</code> can be found in the <a href="https://marshall-lab.org/damidBind/articles/damidBind_vignette.html" class="external-link">damidBind vignette</a>.</p>
</div>
<div class="section level1">
<h1 id="quick-start-guide">Quick start guide<a class="anchor" aria-label="anchor" href="#quick-start-guide"></a>
</h1>
<p>Using <code>damidBind</code>, in eight easy steps:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load up the data </span></span>
<span><span class="co"># (use load_data_genes() for RNA Polymerase occupancy data)</span></span>
<span><span class="va">input</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/load_data_peaks.html">load_data_peaks</a></span><span class="op">(</span></span>
<span>    binding_profiles_path <span class="op">=</span> <span class="st">"path/to/binding_profile_bedgraphs"</span>,</span>
<span>    peaks_path <span class="op">=</span> <span class="st">"path/to/peak_gffs_or_beds"</span></span>
<span>    <span class="co"># add quantile_norm = TRUE if appropriate</span></span>
<span><span class="op">)</span> </span>
<span></span>
<span><span class="co"># Determine differential binding </span></span>
<span><span class="co"># (use differential_accessibility() for CATaDa chromatin accessibility data)</span></span>
<span><span class="va">input.diff</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/differential_binding.html">differential_binding</a></span><span class="op">(</span></span>
<span>    <span class="va">input</span>,</span>
<span>    cond <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>        <span class="st">"Display name 1"</span> <span class="op">=</span> <span class="st">"Condition 1 identifying string in filenames"</span>,</span>
<span>        <span class="st">"Display name 2"</span> <span class="op">=</span> <span class="st">"Condition 2 identifying string in filenames"</span></span>
<span>    <span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># The result 'input.diff' is a formal S4 object.</span></span>
<span><span class="co"># You can see a summary by typing its name:</span></span>
<span><span class="va">input.diff</span></span>
<span></span>
<span><span class="co"># View the proportion of differentially bound loci</span></span>
<span><span class="fu"><a href="reference/plot_venn.html">plot_venn</a></span><span class="op">(</span><span class="va">input.diff</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot the differential binding, labelling associated genes with outliers</span></span>
<span><span class="fu"><a href="reference/plot_volcano.html">plot_volcano</a></span><span class="op">(</span><span class="va">input.diff</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Analyse GO enrichment in peaks associated with one condition</span></span>
<span><span class="fu"><a href="reference/analyse_go_enrichment.html">analyse_go_enrichment</a></span><span class="op">(</span></span>
<span>    <span class="va">input.diff</span>,</span>
<span>    direction <span class="op">=</span> <span class="st">"Condition 1 identifier set with differential_binding() above"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># View the differentially bound loci in an Shiny/IGV browser web app, </span></span>
<span><span class="co"># with an interactive, searchable, sortable table of bound regions</span></span>
<span><span class="fu"><a href="reference/browse_igv_regions.html">browse_igv_regions</a></span><span class="op">(</span><span class="va">input.diff</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Apply additional functions on the differential binding results</span></span>
<span><span class="fu">my_custom_function</span><span class="op">(</span><span class="fu"><a href="reference/analysisTable.html">analysisTable</a></span><span class="op">(</span><span class="va">input.diff</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>

  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://www.bioconductor.org/packages/damidBind" class="external-link">View on Bioconductor</a></li>
<li><a href="https://github.com/marshall-lab/damidBind/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/marshall-lab/damidBind/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/GPL-3" class="external-link">GPL-3</a></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing damidBind</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Owen Marshall <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0003-1605-3871" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a>  </li>
</ul>
</div>



  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Owen Marshall.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
