<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Package index â€¢ damidBind</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/Source_Sans_Pro-0.4.10/font.css" rel="stylesheet"><link href="../deps/Source_Code_Pro-0.4.10/font.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Package index"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">damidBind</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.7</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/damidBind_vignette.html">damidBind</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/marshall-lab/damidBind/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-index">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Package index</h1>
    </div>

    <div class="section level2">
      <h2 id="all-functions">All functions<a class="anchor" aria-label="anchor" href="#all-functions"></a></h2>




    </div><div class="section level2">




      <dl><dt>

          <code><a href="DamIDResults-class.html">DamIDResults-class</a></code> <code><a href="DamIDResults-class.html">DamIDResults</a></code>

        </dt>
        <dd>The DamIDResults Class</dd>
      </dl><dl><dt>

          <code><a href="analyse_go_enrichment.html">analyse_go_enrichment()</a></code>

        </dt>
        <dd>Perform Gene Ontology (GO) enrichment analysis for differentially bound/expressed regions</dd>
      </dl><dl><dt>

          <code><a href="analysisTable.html">analysisTable()</a></code>

        </dt>
        <dd>Access the differential binding analysis results</dd>
      </dl><dl><dt>

          <code><a href="browse_igv_regions.html">browse_igv_regions()</a></code>

        </dt>
        <dd>Interactive IGV visualisation (Shiny + igvShiny) of differential regions</dd>
      </dl><dl><dt>

          <code><a href="calculate_and_add_fdr.html">calculate_and_add_fdr()</a></code>

        </dt>
        <dd>Calculate and add gene occupancy FDR</dd>
      </dl><dl><dt>

          <code><a href="calculate_occupancy.html">calculate_occupancy()</a></code>

        </dt>
        <dd>Compute occupancy for genomic regions</dd>
      </dl><dl><dt>

          <code><a href="conditionNames.html">conditionNames()</a></code>

        </dt>
        <dd>Access condition name mapping</dd>
      </dl><dl><dt>

          <code><a href="differential_accessibility.html">differential_accessibility()</a></code>

        </dt>
        <dd>Differential accessibility analysis for CATaDa (`NOISeq` based)</dd>
      </dl><dl><dt>

          <code><a href="differential_binding.html">differential_binding()</a></code>

        </dt>
        <dd>Differential binding/expression analysis (`limma`)</dd>
      </dl><dl><dt>

          <code><a href="enrichedCond1.html">enrichedCond1()</a></code>

        </dt>
        <dd>Access Condition 1 enriched regions</dd>
      </dl><dl><dt>

          <code><a href="enrichedCond2.html">enrichedCond2()</a></code>

        </dt>
        <dd>Access Condition 2 enriched regions</dd>
      </dl><dl><dt>

          <code><a href="expressed.html">expressed()</a></code>

        </dt>
        <dd>Get expressed genes/loci by FDR</dd>
      </dl><dl><dt>

          <code><a href="filter_genes_by_fdr.html">filter_genes_by_fdr()</a></code>

        </dt>
        <dd>Filter genes by FDR within a specific condition</dd>
      </dl><dl><dt>

          <code><a href="get_ensdb_genes.html">get_ensdb_genes()</a></code>

        </dt>
        <dd>Extract gene annotation from Ensembl via AnnotationHub EnsDb</dd>
      </dl><dl><dt>

          <code><a href="inputData.html">inputData()</a></code>

        </dt>
        <dd>Access original input data and metadata</dd>
      </dl><dl><dt>

          <code><a href="load_data_genes.html">load_data_genes()</a></code>

        </dt>
        <dd>Load genome-wide binding data for gene expression (RNA polymerase occupancy)</dd>
      </dl><dl><dt>

          <code><a href="load_data_peaks.html">load_data_peaks()</a></code>

        </dt>
        <dd>Load genome-wide binding data and associated peak files or GRanges objects</dd>
      </dl><dl><dt>

          <code><a href="plot_venn.html">plot_venn()</a></code>

        </dt>
        <dd>Draw proportional Venn diagrams for differential binding analysis</dd>
      </dl><dl><dt>

          <code><a href="plot_volcano.html">plot_volcano()</a></code>

        </dt>
        <dd>Volcano plot of differentially bound/expressed loci</dd>
      </dl><dl><dt>

          <code><a href="quantile_normalisation.html">quantile_normalisation()</a></code>

        </dt>
        <dd>Quantile normalisation (native R code version)</dd>
      </dl><dl><dt>

          <code><a href="reduce_regions.html">reduce_regions()</a></code>

        </dt>
        <dd>Reduce a list of GRanges to unique, non-overlapping regions</dd>
      </dl><dl><dt>

          <code><a href="sample_labels_by_isolation.html">sample_labels_by_isolation()</a></code>

        </dt>
        <dd>Sample data points based on local isolation</dd>
      </dl></div>
  </main></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Owen Marshall.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer></div>





  </body></html>

