<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Load genome-wide binding data for gene expression (RNA polymerase occupancy) — load_data_genes • damidBind</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/Source_Sans_Pro-0.4.10/font.css" rel="stylesheet"><link href="../deps/Source_Code_Pro-0.4.10/font.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Load genome-wide binding data for gene expression (RNA polymerase occupancy) — load_data_genes"><meta name="description" content="Reads RNA Polymerase DamID binding profiles either from bedGraph files or
directly from a named list of GRanges objects. Calculates binding occupancy
summarised over genes."><meta property="og:description" content="Reads RNA Polymerase DamID binding profiles either from bedGraph files or
directly from a named list of GRanges objects. Calculates binding occupancy
summarised over genes."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">damidBind</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.10</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/damidBind_vignette.html">damidBind</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/marshall-lab/damidBind/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Load genome-wide binding data for gene expression (RNA polymerase occupancy)</h1>
      <small class="dont-index">Source: <a href="https://github.com/marshall-lab/damidBind/blob/HEAD/R/load_data.R" class="external-link"><code>R/load_data.R</code></a></small>
      <div class="d-none name"><code>load_data_genes.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Reads RNA Polymerase DamID binding profiles either from bedGraph files or
directly from a named list of GRanges objects. Calculates binding occupancy
summarised over genes.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">load_data_genes</span><span class="op">(</span></span>
<span>  binding_profiles_path <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  binding_profiles <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  drop_samples <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  quantile_norm <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  organism <span class="op">=</span> <span class="st">"drosophila melanogaster"</span>,</span>
<span>  calculate_fdr <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  fdr_iterations <span class="op">=</span> <span class="fl">50000</span>,</span>
<span>  ensdb_genes <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  BPPARAM <span class="op">=</span> <span class="fu">BiocParallel</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocParallel/man/register.html" class="external-link">bpparam</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  plot_diagnostics <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/interactive.html" class="external-link">interactive</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-binding-profiles-path">binding_profiles_path<a class="anchor" aria-label="anchor" href="#arg-binding-profiles-path"></a></dt>
<dd><p>Character vector of directories or file globs
containing log2 ratio binding tracks in bedGraph format. Wildcards ('*') supported.</p></dd>


<dt id="arg-binding-profiles">binding_profiles<a class="anchor" aria-label="anchor" href="#arg-binding-profiles"></a></dt>
<dd><p>Named list of GRanges objects representing binding profiles.</p></dd>


<dt id="arg-drop-samples">drop_samples<a class="anchor" aria-label="anchor" href="#arg-drop-samples"></a></dt>
<dd><p>A character vector of sample names or patterns to remove.
Matching samples are removed from the analysis before normalisation and
occupancy calculation. This can be useful for excluding samples that fail
initial quality checks. Default: `NULL` (no samples are dropped).</p></dd>


<dt id="arg-quantile-norm">quantile_norm<a class="anchor" aria-label="anchor" href="#arg-quantile-norm"></a></dt>
<dd><p>Logical (default: FALSE) quantile-normalise across all
signal columns if TRUE.</p></dd>


<dt id="arg-organism">organism<a class="anchor" aria-label="anchor" href="#arg-organism"></a></dt>
<dd><p>Organism string (lower case) to obtain genome annotation from (if not providing a custom `ensdb_genes` object)
Defautls to "drosophila melanogaster".</p></dd>


<dt id="arg-calculate-fdr">calculate_fdr<a class="anchor" aria-label="anchor" href="#arg-calculate-fdr"></a></dt>
<dd><p>Calculate FDR based on RNA Pol occupancy (see details) (default: FALSE)</p></dd>


<dt id="arg-fdr-iterations">fdr_iterations<a class="anchor" aria-label="anchor" href="#arg-fdr-iterations"></a></dt>
<dd><p>Number of iterations to use to determine null model for FDR (default: 50000)</p></dd>


<dt id="arg-ensdb-genes">ensdb_genes<a class="anchor" aria-label="anchor" href="#arg-ensdb-genes"></a></dt>
<dd><p>GRanges object: gene annotation. Automatically obtained
from `organism` if NULL.</p></dd>


<dt id="arg-bpparam">BPPARAM<a class="anchor" aria-label="anchor" href="#arg-bpparam"></a></dt>
<dd><p>BiocParallel function (defaults to BiocParallel::bpparam())</p></dd>


<dt id="arg-plot-diagnostics">plot_diagnostics<a class="anchor" aria-label="anchor" href="#arg-plot-diagnostics"></a></dt>
<dd><p>Logical. If `TRUE` (the default in interactive sessions),
diagnostic plots (PCA and correlation heatmap) will be generated and
displayed for both the raw binding data and the summarised occupancy data.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>List with elements:</p>
<dl><dt>binding_profiles_data</dt>
<dd><p>data.frame of merged binding profiles, with chr, start, end, sample columns.</p></dd>

  <dt>occupancy</dt>
<dd><p>data.frame of occupancy values summarised over genes.</p></dd>

  <dt>test_category</dt>
<dd><p>Character scalar; will be "expressed".</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>One of `binding_profiles_path` or `binding_profiles` must be provided.</p>
<p>When supplying GRanges lists, each GRanges should contain exactly one numeric
metadata column representing the signal, and `binding_profiles` must be a
named list, with element names used as sample names.</p>

<p>The algorithm for determining gene occupancy FDR (as a proxy for gene expression)
is based on `polii.gene.call`, which in turn was based on that described in
Southall et al. (2013). Dev Cell, 26(1), 101–12. doi:10.1016/j.devcel.2013.05.020.
Briefly, the algorithm establishes a null model by simulating the distribution of mean occupancy scores
from random fragments.  It fits a two-tiered regression to predict the False Discovery Rate (FDR), based
on fragment count and score. For each gene, the true weighted mean occupancy and fragment count are
calculated from the provided binding profile. Finally, the pre-computed regression models are used
to assign a specific FDR to each gene based on its observed occupancy and fragment count.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># Create a mock GRanges object for gene annotations</span></span></span>
<span class="r-in"><span><span class="co"># This object, based on the package's unit tests, avoids network access</span></span></span>
<span class="r-in"><span><span class="co"># and includes a very long gene to ensure overlaps with sample data.</span></span></span>
<span class="r-in"><span><span class="va">mock_genes_gr</span> <span class="op">&lt;-</span> <span class="fu">GenomicRanges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    seqnames <span class="op">=</span> <span class="fu">S4Vectors</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Rle-class.html" class="external-link">Rle</a></span><span class="op">(</span><span class="st">"2L"</span>, <span class="fl">7</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    ranges <span class="op">=</span> <span class="fu">IRanges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html" class="external-link">IRanges</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>        start <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1000</span>, <span class="fl">2000</span>, <span class="fl">3000</span>, <span class="fl">5000</span>, <span class="fl">6000</span>, <span class="fl">7000</span>, <span class="fl">8000</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>        end <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1500</span>, <span class="fl">2500</span>, <span class="fl">3500</span>, <span class="fl">5500</span>, <span class="fl">6500</span>, <span class="fl">7500</span>, <span class="fl">20000000</span><span class="op">)</span></span></span>
<span class="r-in"><span>    <span class="op">)</span>,</span></span>
<span class="r-in"><span>    strand <span class="op">=</span> <span class="fu">S4Vectors</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Rle-class.html" class="external-link">Rle</a></span><span class="op">(</span><span class="fu">GenomicRanges</span><span class="fu">::</span><span class="fu">strand</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"+"</span>, <span class="st">"-"</span>, <span class="st">"+"</span>, <span class="st">"+"</span>, <span class="st">"-"</span>, <span class="st">"-"</span>, <span class="st">"+"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    gene_id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"FBgn001"</span>, <span class="st">"FBgn002"</span>, <span class="st">"FBgn003"</span>, <span class="st">"FBgn004"</span>, <span class="st">"FBgn005"</span>, <span class="st">"FBgn006"</span>, <span class="st">"FBgn007"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    gene_name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"geneA"</span>, <span class="st">"geneB"</span>, <span class="st">"geneC"</span>, <span class="st">"geneD"</span>, <span class="st">"geneE"</span>, <span class="st">"geneF"</span>, <span class="st">"LargeTestGene"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Get path to sample data files included with the package</span></span></span>
<span class="r-in"><span><span class="va">data_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, package <span class="op">=</span> <span class="st">"damidBind"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Run loading function using sample files and mock gene annotations</span></span></span>
<span class="r-in"><span><span class="co"># This calculates occupancy over genes instead of peaks.</span></span></span>
<span class="r-in"><span><span class="va">loaded_data_genes</span> <span class="op">&lt;-</span> <span class="fu">load_data_genes</span><span class="op">(</span></span></span>
<span class="r-in"><span>    binding_profiles_path <span class="op">=</span> <span class="va">data_dir</span>,</span></span>
<span class="r-in"><span>    ensdb_genes <span class="op">=</span> <span class="va">mock_genes_gr</span>,</span></span>
<span class="r-in"><span>    quantile_norm <span class="op">=</span> <span class="cn">FALSE</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Locating binding profile files</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Building binding profile dataframe from input files ...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Loaded: Bsh_Dam_L4_r1-ext300-vs-Dam.kde-norm</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Loaded: Bsh_Dam_L4_r2-ext300-vs-Dam.kde-norm</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Loaded: Bsh_Dam_L5_r1-ext300-vs-Dam.kde-norm</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  - Loaded: Bsh_Dam_L5_r2-ext300-vs-Dam.kde-norm</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Calculating average occupancy for 7 regions...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Generating diagnostic plots...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   3 rows with zero variance were filtered.</span>
<span class="r-plt img"><img src="load_data_genes-1.png" alt="" width="700" height="433"></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   3989 rows with zero variance were filtered.</span>
<span class="r-plt img"><img src="load_data_genes-2.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># View the head of the occupancy table</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">loaded_data_genes</span><span class="op">$</span><span class="va">occupancy</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>              gene_id gene_name         name nfrags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2L:1000-1500 FBgn001     geneA 2L:1000-1500      4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2L:2000-2500 FBgn002     geneB 2L:2000-2500      4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2L:3000-3500 FBgn003     geneC 2L:3000-3500      4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2L:5000-5500 FBgn004     geneD 2L:5000-5500      4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2L:6000-6500 FBgn005     geneE 2L:6000-6500      2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2L:7000-7500 FBgn006     geneF 2L:7000-7500      1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>              Bsh_Dam_L4_r1-ext300-vs-Dam.kde-norm</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2L:1000-1500                             0.000000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2L:2000-2500                             0.000000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2L:3000-3500                             0.000000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2L:5000-5500                             1.465469</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2L:6000-6500                             1.515389</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2L:7000-7500                            -0.130000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>              Bsh_Dam_L4_r2-ext300-vs-Dam.kde-norm</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2L:1000-1500                             0.000000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2L:2000-2500                             0.000000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2L:3000-3500                             0.000000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2L:5000-5500                             1.444371</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2L:6000-6500                             1.791916</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2L:7000-7500                            -0.110000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>              Bsh_Dam_L5_r1-ext300-vs-Dam.kde-norm</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2L:1000-1500                            0.0000000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2L:2000-2500                            0.0000000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2L:3000-3500                            0.0000000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2L:5000-5500                            1.2137325</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2L:6000-6500                            0.2373453</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2L:7000-7500                           -0.0300000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>              Bsh_Dam_L5_r2-ext300-vs-Dam.kde-norm</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2L:1000-1500                            0.0000000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2L:2000-2500                            0.0000000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2L:3000-3500                            0.0000000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2L:5000-5500                            0.8943313</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2L:6000-6500                            0.6489222</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2L:7000-7500                           -0.0900000</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Owen Marshall.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer></div>





  </body></html>

