<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>damidBind • damidBind</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/Source_Sans_Pro-0.4.10/font.css" rel="stylesheet">
<link href="../deps/Source_Code_Pro-0.4.10/font.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="damidBind">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">damidBind</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.6</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/damidBind_vignette.html">damidBind</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/marshall-lab/damidBind/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>damidBind</h1>
                        <h4 data-toc-skip class="author">Owen J.
Marshall</h4>
            <address class="author_afil">
      Menzies Institute for Medical Research, University of
Tasmania<br><a class="author_email" href="mailto:#"></a><a href="mailto:owen.marshall@utas.edu.au" class="email">owen.marshall@utas.edu.au</a>
      </address>
                  
      
      <small class="dont-index">Source: <a href="https://github.com/marshall-lab/damidBind/blob/HEAD/vignettes/damidBind_vignette.Rmd" class="external-link"><code>vignettes/damidBind_vignette.Rmd</code></a></small>
      <div class="d-none name"><code>damidBind_vignette.Rmd</code></div>
    </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>The damidBind package provides a straightforward formal
      analysis pipeline to analyse and explore differential DamID
      binding, gene transcription or chromatin accessibility between two
      conditions. The package imports processed data from DamID-seq
      experiments, either as external raw files in the form of binding
      bedGraphs and GFF/BED peak calls, or as internal lists of GRanges
      objects. After optionally normalising data, combining peaks across
      replicates and determining per-replicate peak occupancy, the
      package links bound loci to nearby genes. For RNA Polymerase DamID
      data, the package calculates occupancy over genes, and optionally
      calcualates the FDR of significantly-enriched gene occupancy.
      damidBind then uses either limma (for conventional log2 ratio
      DamID binding data) or NOIseq (for counts-based CATaDa chromatin
      accessibility data) to identify differentially-enriched regions,
      or differentially epxressed genes, between two conditions. The
      package provides a number of visualisation tools (volcano plots,
      Gene Ontology enrichment plots via ClusterProfiler and
      proportional Venn diagrams via BioVenn for downstream data
      exploration and analysis. An powerful, interactive IGV genome
      browser interface (powered by Shiny and igvShiny) allows users to
      rapidly and intuitively assess significant differentially-bound
      regions in their genomic context.</p>
    </div>
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>DamID <span class="citation">(van Steensel &amp; Henikoff, 2000; van
Steensel <em>et al</em>, 2001)</span> is a highly-sensitive means to
profile the genome-wide association of proteins with chromatin in living
eukaryotic cells, without fixation or the use of antibodies. Cell-type
specific techniques such as Targeted DamID <span class="citation">(Southall <em>et al</em>, 2013; Marshall <em>et
al</em>, 2016)</span> to profile protein binding, and CATaDA <span class="citation">(Aughey <em>et al</em>, 2018)</span> to profile
chromatin accessibility, have made the technique an extremely powerful
tool to understand the binding of transcription factors, chromatin
proteins, RNA polymerase and chromatin changes during development and
disease.</p>
<p>Despite the technique’s growing popularity and adoption, no formal
analysis pipeline or R package exists to analyse and explore
differential DamID binding, gene transcription or chromatin
accessibility between two conditions. The <em><a href="https://bioconductor.org/packages/3.20/damidBind" class="external-link">damidBind</a></em>
package provides this functionality.</p>
<p><em><a href="https://bioconductor.org/packages/3.20/damidBind" class="external-link">damidBind</a></em>
imports processed data from DamID-seq experiments in the form of binding
bedgraphs and GFF peak calls. After optionally normalising data,
combining peaks across replicates and determining per-replicate peak
occupancy, the package links bound loci to nearby genes. It then uses
either <em><a href="https://bioconductor.org/packages/3.20/limma" class="external-link">limma</a></em> (for
conventional log2 ratio DamID binding data) or <em><a href="https://bioconductor.org/packages/3.20/NOIseq" class="external-link">NOIseq</a></em>
(for counts-based CATaDa chromatin accessibility data) to identify
differentially-enriched regions between two conditions. The package
provides a number of visualisation tools (volcano plots, GO enrichment
plots, Venn diagrams) for downstream data exploration and analysis. An
interactive IGV genome browser interface (powered by <em><a href="https://CRAN.R-project.org/package=Shiny" class="external-link">Shiny</a></em> and
<em><a href="https://bioconductor.org/packages/3.20/igvShiny" class="external-link">igvShiny</a></em>)
allows users to rapidly and intuitively assess significant
differentially-bound regions.</p>
<p>Although extensive customisation options are available if required,
much of the data handling by <em><a href="https://bioconductor.org/packages/3.20/damidBind" class="external-link">damidBind</a></em>
is taken care of automatically, with sensible defaults assumed. To move
from loading raw data to visualising differentially-enriched regions on
a volcano plot or browsing enriched regions in an interactive IGV window
is a simple three command procedure.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>To install from github, use:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Install Bioconductor</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.github.io/BiocManager/" class="external-link">"BiocManager"</a></span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Then install the damidBind package</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"damidBind"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="quick-start-guide">Quick start guide<a class="anchor" aria-label="anchor" href="#quick-start-guide"></a>
</h2>
<p>Using <em><a href="https://bioconductor.org/packages/3.20/damidBind" class="external-link">damidBind</a></em>,
in eight easy steps:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Example code only, not run:</span></span>
<span></span>
<span><span class="co"># Load up the data (use load_data_genes() for RNA Polymerase occupancy data)</span></span>
<span><span class="va">input</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/load_data_peaks.html">load_data_peaks</a></span><span class="op">(</span></span>
<span>    binding_profiles_path <span class="op">=</span> <span class="st">"path/to/binding_profile_bedgraphs"</span>,</span>
<span>    peaks_path <span class="op">=</span> <span class="st">"path/to/peak_gffs_or_beds"</span></span>
<span><span class="op">)</span> <span class="co"># add quantile_norm = TRUE if appropriate</span></span>
<span></span>
<span><span class="co"># Deterimine differential binding  (use differential_accesibility() for CATaDa chromatin accessibility data)</span></span>
<span><span class="va">input.diff</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/differential_binding.html">differential_binding</a></span><span class="op">(</span></span>
<span>    <span class="va">input</span>,</span>
<span>    cond <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>        <span class="st">"Condition 1 identifying string in filenames"</span>,</span>
<span>        <span class="st">"Condition 2 identifying string in filenames"</span></span>
<span>    <span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># The result 'input.diff' is a formal S4 object.</span></span>
<span><span class="co"># You can see a summary by simply typing its name:</span></span>
<span><span class="va">input.diff</span></span>
<span></span>
<span><span class="co"># View the proporition of differentially bound loci</span></span>
<span><span class="fu"><a href="../reference/plot_venn.html">plot_venn</a></span><span class="op">(</span><span class="va">input.diff</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot the differential binding, labelling associated genes with outliers</span></span>
<span><span class="fu"><a href="../reference/plot_volcano.html">plot_volcano</a></span><span class="op">(</span><span class="va">input.diff</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Analyse GO enrichment in peaks associated with one condition</span></span>
<span><span class="fu"><a href="../reference/analyse_go_enrichment.html">analyse_go_enrichment</a></span><span class="op">(</span></span>
<span>    <span class="va">input.diff</span>,</span>
<span>    direction <span class="op">=</span> <span class="st">"Condition 1 identifier set with differential_binding() above"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># View the differentially bound regions in an IGV browser window, with an interactive table of bound regions</span></span>
<span><span class="fu"><a href="../reference/browse_igv_regions.html">browse_igv_regions</a></span><span class="op">(</span><span class="va">input.diff</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Apply additional functions on the differential binding results</span></span>
<span><span class="fu">my_custom_function</span><span class="op">(</span><span class="fu"><a href="../reference/analysisTable.html">analysisTable</a></span><span class="op">(</span><span class="va">input.diff</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="sample-data-examples">Sample data examples<a class="anchor" aria-label="anchor" href="#sample-data-examples"></a>
</h2>
<div class="section level3">
<h3 id="sample-input-data-provided-within-the-package">Sample input data (provided within the package)<a class="anchor" aria-label="anchor" href="#sample-input-data-provided-within-the-package"></a>
</h3>
<p><em><a href="https://bioconductor.org/packages/3.20/damidBind" class="external-link">damidBind</a></em>
provides a simple, truncated dataset with the package for immediate
testing. Owing to package space, this sample dataset uses only
<em>Drosophila melanogaster</em> chromosome 2L binding, and should not
be used for any actual analysis.</p>
</div>
<div class="section level3">
<h3 id="sample-input-data-provided-online-through-a-zenodo-repository">Sample input data (provided online through a Zenodo repository)<a class="anchor" aria-label="anchor" href="#sample-input-data-provided-online-through-a-zenodo-repository"></a>
</h3>
<p><a href="https://doi.org/10.5281/zenodo.16649477" class="external-link">Three
previously-published sample datasets</a> have been deposited in Zenodo
and made available for fully exploring the features and capabilities of
<em><a href="https://bioconductor.org/packages/3.20/damidBind" class="external-link">damidBind</a></em>:</p>
<ul>
<li>Bsh transcription factor binding in L4 and L5 neurons of the
<em>Drosophila</em> lamina <span class="citation">(Xu <em>et al</em>,
2024)</span>
</li>
<li>CATaDa chromatin accessibility data in L4 and L5 neurons <span class="citation">(Xu <em>et al</em>, 2024)</span>
</li>
<li>RNA Polymerase II occupancy in <em>Drosophila</em> larval neural
stem cells and adult neurons. <span class="citation">(Marshall &amp;
Brand, 2017)</span>
</li>
</ul>
<p>Processed files (incl. binding profiles and peaks files) are provided
for all datasets.</p>
</div>
</div>
<div class="section level2">
<h2 id="data-preparation">Data preparation<a class="anchor" aria-label="anchor" href="#data-preparation"></a>
</h2>
<div class="section level3">
<h3 id="input-data-format">Input data format<a class="anchor" aria-label="anchor" href="#input-data-format"></a>
</h3>
<p><em><a href="https://bioconductor.org/packages/3.20/damidBind" class="external-link">damidBind</a></em>
takes as input:</p>
<ul>
<li>Genome-wide binding profile tracks in BEDGRAPH format (external
files) or <em><a href="https://bioconductor.org/packages/3.20/GenomicRanges" class="external-link">GenomicRanges</a></em>
format (internal)
<ul>
<li>For standard DamID binding datasets, the score column is expected to
be a log2 ratio</li>
<li>For CATaDa chromatin accessibility datasets, the score column should
be in CPM (counts per million reads) or similar</li>
</ul>
</li>
<li>Significant peaks files in GFF or BED format (external files) or
<em><a href="https://bioconductor.org/packages/3.20/GenomicRanges" class="external-link">GenomicRanges</a></em>
format (internal)
<ul>
<li>Peaks are not required when comparing differential gene expression
via RNA Polymerase occupancy.</li>
</ul>
</li>
</ul>
<p>All external data files are read using <em><a href="https://bioconductor.org/packages/3.20/rtracklayer" class="external-link">rtracklayer</a></em>
and can be gzip compressed or uncompressed as required.</p>
</div>
<div class="section level3">
<h3 id="input-data-generation">Input data generation<a class="anchor" aria-label="anchor" href="#input-data-generation"></a>
</h3>
<p><em><a href="https://bioconductor.org/packages/3.20/damidBind" class="external-link">damidBind</a></em>
can work with either raw data files or preprocessed data in <em><a href="https://bioconductor.org/packages/3.20/GenomicRanges" class="external-link">GenomicRanges</a></em>
format.</p>
<div class="section level4">
<h4 id="bedgraph-binding-accessibility-profiles">BEDGRAPH binding / accessibility profiles<a class="anchor" aria-label="anchor" href="#bedgraph-binding-accessibility-profiles"></a>
</h4>
<p>We recommend that <a href="https://owenjm.github.io/damidseq_pipeline" class="external-link">damidseq_pipeline</a>
is used to generate input files:</p>
<ul>
<li>Using default options to generate log2 ratio protein binding
BEDGRAPHs</li>
<li>Using the <code>--catada</code> flag on Dam-only samples to generate
count-based CATaDa accesibility BEDGRAPHs</li>
</ul>
</div>
<div class="section level4">
<h4 id="peak-gff-files">Peak GFF files<a class="anchor" aria-label="anchor" href="#peak-gff-files"></a>
</h4>
<p>We recommend that <a href="https://github.com/owenjm/find_peaks" class="external-link">find_peaks</a> is used to
generate peak files on each DamID / CATaDa BEDGRAPH file replicate:</p>
<ul>
<li>In all cases, the default options of
<code>--min_quant=0.8 --unified_peaks=min</code> are recommended</li>
</ul>
</div>
</div>
</div>
<div class="section level2">
<h2 id="using-damidbind">Using damidBind<a class="anchor" aria-label="anchor" href="#using-damidbind"></a>
</h2>
<p>Load the library</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://marshall-lab.org/damidBind" class="external-link">damidBind</a></span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span></code></pre>
<div class="section level3">
<h3 id="loading-data">Loading data<a class="anchor" aria-label="anchor" href="#loading-data"></a>
</h3>
<p>A small example dataset is provided with <em><a href="https://bioconductor.org/packages/3.20/damidBind" class="external-link">damidBind</a></em>.
The dataset is derived from the binding of Bsh in L4 and L5 neurons of
the <em>Drosophila melanogaster</em> lamina <span class="citation">(Xu
<em>et al</em>, 2024)</span>, truncated to chromosome 2L. The raw data
was processed using <code>damidseq_pipeline</code> and peaks called
using <code>find_peaks</code> (see the publication Methods for more
details).</p>
<p>To load up the data, we only need to know the path to the installed
example datafiles:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, package <span class="op">=</span> <span class="st">"damidBind"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Show the files present for clarity in this vignette example:</span></span>
<span><span class="va">files</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">list.files</a></span><span class="op">(</span><span class="va">data_dir</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">files</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "Bsh_Dam_L4_r1-ext300-vs-Dam.kde-norm.gatc-FDR0.01.peaks.2L.bed.gz"</span></span>
<span><span class="co">## [2] "Bsh_Dam_L4_r1-ext300-vs-Dam.kde-norm.gatc.2L.bedgraph.gz"         </span></span>
<span><span class="co">## [3] "Bsh_Dam_L4_r2-ext300-vs-Dam.kde-norm.gatc-FDR0.01.peaks.2L.bed.gz"</span></span>
<span><span class="co">## [4] "Bsh_Dam_L4_r2-ext300-vs-Dam.kde-norm.gatc.2L.bedgraph.gz"         </span></span>
<span><span class="co">## [5] "Bsh_Dam_L5_r1-ext300-vs-Dam.kde-norm.gatc-FDR0.01.peaks.2L.bed.gz"</span></span>
<span><span class="co">## [6] "Bsh_Dam_L5_r1-ext300-vs-Dam.kde-norm.gatc.2L.bedgraph.gz"         </span></span>
<span><span class="co">## [7] "Bsh_Dam_L5_r2-ext300-vs-Dam.kde-norm.gatc-FDR0.01.peaks.2L.bed.gz"</span></span>
<span><span class="co">## [8] "Bsh_Dam_L5_r2-ext300-vs-Dam.kde-norm.gatc.2L.bedgraph.gz"</span></span></code></pre>
<p>And then we can use <em><a href="https://bioconductor.org/packages/3.20/damidBind" class="external-link">damidBind</a></em>
on the data. First, we load up the data. There are two potential
commands that can be used for data loading, depending on the data being
analysed:</p>
<dl>
<dt><code><a href="../reference/load_data_peaks.html">load_data_peaks()</a></code></dt>
<dd>
<p>Used to load binding data with associated peaks files
(e.g. transcription factor binding, CATaDa accessibility)</p>
</dd>
<dt><code><a href="../reference/load_data_genes.html">load_data_genes()</a></code></dt>
<dd>
<p>Used to load RNA Polymerase occupancy DamID data (as a proxy for gene
expression, as per <span class="citation">Marshall &amp; Brand
(2015)</span>). For these data, no peak files are required (as occupancy
is calculated over the gene bodies).</p>
</dd>
</dl>
<p>In this case, we’re dealing with the binding of the transcription
factor Bsh, so we’ll use the <code><a href="../reference/load_data_peaks.html">load_data_peaks()</a></code>. As this is
the same transcription factor being profiled in two different cell
types, and we’d expect the binding distribution to be similar, we’ll use
quantile normalisation on the datasets.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">input.bsh</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/load_data_peaks.html">load_data_peaks</a></span><span class="op">(</span></span>
<span>    binding_profiles_path <span class="op">=</span> <span class="va">data_dir</span>,</span>
<span>    peaks_path <span class="op">=</span> <span class="va">data_dir</span>,</span>
<span>    quantile_norm <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Finding genome versions ...</span></span></code></pre>
<pre><code><span><span class="co">## Loading Ensembl genome version 'Ensembl 113 EnsDb for Drosophila melanogaster'</span></span></code></pre>
<pre><code><span><span class="co">## loading from cache</span></span></code></pre>
<pre><code><span><span class="co">## require("ensembldb")</span></span></code></pre>
<pre><code><span><span class="co">## Locating binding profile files</span></span></code></pre>
<pre><code><span><span class="co">## Building binding profile dataframe from input files ...</span></span></code></pre>
<pre><code><span><span class="co">##  - Loaded: Bsh_Dam_L4_r1-ext300-vs-Dam.kde-norm</span></span></code></pre>
<pre><code><span><span class="co">##  - Loaded: Bsh_Dam_L4_r2-ext300-vs-Dam.kde-norm</span></span></code></pre>
<pre><code><span><span class="co">##  - Loaded: Bsh_Dam_L5_r1-ext300-vs-Dam.kde-norm</span></span></code></pre>
<pre><code><span><span class="co">##  - Loaded: Bsh_Dam_L5_r2-ext300-vs-Dam.kde-norm</span></span></code></pre>
<pre><code><span><span class="co">## Applying quantile normalisation</span></span></code></pre>
<pre><code><span><span class="co">## Locating peak files</span></span></code></pre>
<pre><code><span><span class="co">## Calculating occupancy over peaks</span></span></code></pre>
<pre><code><span><span class="co">## Calculating average occupancy for 1208 regions...</span></span></code></pre>
<div class="section level4">
<h4 id="loading-data-from-granges-objects">Loading data from GRanges objects<a class="anchor" aria-label="anchor" href="#loading-data-from-granges-objects"></a>
</h4>
<p>In some workflows, you may already have your binding profiles and
peak regions loaded as <em><a href="https://bioconductor.org/packages/3.20/GenomicRanges" class="external-link">GenomicRanges</a></em>
objects. <code>damidBind</code> supports this directly, bypassing the
need to read from file paths.</p>
<p>The <code><a href="../reference/load_data_peaks.html">load_data_peaks()</a></code> and <code><a href="../reference/load_data_genes.html">load_data_genes()</a></code>
functions accept <code>GRanges</code> objects via the
<code>binding_profiles</code> and <code>peaks</code> arguments, when
used in place of <code>binding_profiles_paths</code> and
<code>peaks_paths</code>. These inputs must be provided as <strong>named
lists</strong> of <code>GRanges</code> objects, where the names
correspond to your sample identifiers. For binding profiles, each
<code>GRanges</code> object must contain exactly one numeric metadata
column (e.g., <code>score</code>) representing the signal.</p>
<p>The following example first loads the provided sample data files to
generate <code>GRanges</code> objects, and then loads these into
<code>damidBind</code> to generate an input object ready for downstream
analysis.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Locate bedGraph and peaks example files</span></span>
<span><span class="va">bedgraph_files</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">list.files</a></span><span class="op">(</span><span class="va">data_dir</span>, pattern <span class="op">=</span> <span class="st">"\\.bedgraph\\.gz$"</span>, full.names <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">peak_files</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">list.files</a></span><span class="op">(</span><span class="va">data_dir</span>, pattern <span class="op">=</span> <span class="st">"\\.bed\\.gz$"</span>, full.names <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Obtain unique sample names from the filenames (specific to these example files)</span></span>
<span><span class="va">sample_names</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"-ext300-vs-Dam.kde-norm.gatc.*"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/r/base/basename.html" class="external-link">basename</a></span><span class="op">(</span><span class="va">bedgraph_files</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Load the bedgraph files into a named list of GRanges objects</span></span>
<span><span class="va">binding_gr_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">bedgraph_files</span>, <span class="fu">rtracklayer</span><span class="fu">::</span><span class="va">import</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">binding_gr_list</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">sample_names</span></span>
<span></span>
<span><span class="co"># Similarly, load the peak files into a named list of GRanges objects</span></span>
<span><span class="va">peak_gr_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">peak_files</span>, <span class="fu">rtracklayer</span><span class="fu">::</span><span class="va">import</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">peak_gr_list</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">sample_names</span></span>
<span></span>
<span><span class="co"># Now, call load_data_peaks() using the GRanges lists instead of file paths</span></span>
<span><span class="va">input.bsh_from_gr</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/load_data_peaks.html">load_data_peaks</a></span><span class="op">(</span></span>
<span>    binding_profiles <span class="op">=</span> <span class="va">binding_gr_list</span>,</span>
<span>    peaks <span class="op">=</span> <span class="va">peak_gr_list</span>,</span>
<span>    quantile_norm <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Finding genome versions ...</span></span></code></pre>
<pre><code><span><span class="co">## Loading Ensembl genome version 'Ensembl 113 EnsDb for Drosophila melanogaster'</span></span></code></pre>
<pre><code><span><span class="co">## loading from cache</span></span></code></pre>
<pre><code><span><span class="co">## Building binding profile dataframe from supplied GRanges objects ...</span></span></code></pre>
<pre><code><span><span class="co">## Applying quantile normalisation</span></span></code></pre>
<pre><code><span><span class="co">## Using supplied peaks GRanges list.</span></span></code></pre>
<pre><code><span><span class="co">## Calculating occupancy over peaks</span></span></code></pre>
<pre><code><span><span class="co">## Calculating average occupancy for 1208 regions...</span></span></code></pre>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># The resulting object can now be used for differential analysis,</span></span>
<span><span class="co"># just as in the examples below.</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="gene-locus-assignment-to-peaks">Gene locus assignment to peaks<a class="anchor" aria-label="anchor" href="#gene-locus-assignment-to-peaks"></a>
</h4>
<p><code>damidBind</code> automatically assigns peaks to genes within
1kb of the peak boundary. This value (in bp) can be changed using the
<code>maxgap_loci</code> parameter of <code><a href="../reference/load_data_peaks.html">load_data_peaks()</a></code>.
Note that more than one gene locus can be assigned to a peak (especially
within the gene-dense <em>Drosophila</em> genome).</p>
</div>
</div>
<div class="section level3">
<h3 id="analysing-differential-binding">Analysing differential binding<a class="anchor" aria-label="anchor" href="#analysing-differential-binding"></a>
</h3>
<p>Now, we’ll determine the differential binding of Bsh between L4 and
L5 neurons. Our input files use <code>L4</code> and <code>L5</code>
labels to distinguish the samples, and that’s all we need to know
here.</p>
<p>Again, there are two different analysis options depending on the data
type:</p>
<dl>
<dt><code><a href="../reference/differential_binding.html">differential_binding()</a></code></dt>
<dd>
<p>Analyses conventional DamID data (including RNA Polymerase DamID)
present as a log2(Dam-fusion/Dam-only) ratio. All standard DamID data is
in this format. <em><a href="https://bioconductor.org/packages/3.20/limma" class="external-link">limma</a></em> is
used as the backend for analysis.</p>
</dd>
<dt><code><a href="../reference/differential_accessibility.html">differential_accessibility()</a></code></dt>
<dd>
<p>Analyses CATaDa data (Dam-only chromatin accessibility data) present
as Counts Per Million reads (CPM) or raw counts. Only CATaDa data
(generated via e.g. <code>damidseq_pipeline --catada</code>) should be
used with this function. Given the counts nature of these data, <em><a href="https://bioconductor.org/packages/3.20/NOISeq" class="external-link">NOISeq</a></em> is
used as the analysis backend.</p>
</dd>
</dl>
<p>The input from <code><a href="../reference/load_data_peaks.html">load_data_peaks()</a></code> or
<code><a href="../reference/load_data_genes.html">load_data_genes()</a></code> feeds directly into these functions.</p>
<p>Here, we’re not dealing with CATaDa data, so we’ll use
<code><a href="../reference/differential_binding.html">differential_binding()</a></code> to find the significant,
differentially-bound loci between the two conditions:</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">diff.bsh</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/differential_binding.html">differential_binding</a></span><span class="op">(</span></span>
<span>    <span class="va">input.bsh</span>,</span>
<span>    cond <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"L4 neurons"</span> <span class="op">=</span> <span class="st">"L4"</span>,</span>
<span>             <span class="st">"L5 neurons"</span> <span class="op">=</span> <span class="st">"L5"</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Condition display names were sanitized for internal data:</span></span></code></pre>
<pre><code><span><span class="co">##   'L4 neurons' -&gt; 'L4.neurons'</span></span></code></pre>
<pre><code><span><span class="co">##   'L5 neurons' -&gt; 'L5.neurons'</span></span></code></pre>
<pre><code><span><span class="co">## Applying filter: Loci must have occupancy &gt; 0 in at least 2 samples of at least one condition.</span></span></code></pre>
<pre><code><span><span class="co">##   Filtered out 12 loci. 1196 loci remain for analysis.</span></span></code></pre>
<pre><code><span><span class="co">## Differential analysis setup:</span></span></code></pre>
<pre><code><span><span class="co">## Condition 1 Display Name: 'L4 neurons' (Internal: 'L4.neurons', Match Pattern: 'L4')</span></span></code></pre>
<pre><code><span><span class="co">##   Found 2 replicates:</span></span>
<span><span class="co">##     Bsh_Dam_L4_r1-ext300-vs-Dam.kde-norm_qnorm</span></span>
<span><span class="co">##     Bsh_Dam_L4_r2-ext300-vs-Dam.kde-norm_qnorm</span></span></code></pre>
<pre><code><span><span class="co">## Condition 2 Display Name: 'L5 neurons' (Internal: 'L5.neurons', Match Pattern: 'L5')</span></span></code></pre>
<pre><code><span><span class="co">##   Found 2 replicates:</span></span>
<span><span class="co">##     Bsh_Dam_L5_r1-ext300-vs-Dam.kde-norm_qnorm</span></span>
<span><span class="co">##     Bsh_Dam_L5_r2-ext300-vs-Dam.kde-norm_qnorm</span></span></code></pre>
<pre><code><span><span class="co">## limma contrasts: L4.neurons-L5.neurons</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Filtering for positive enrichment (mean score &gt; 0 in the enriched condition).</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## 284 loci enriched in L4 neurons</span></span></code></pre>
<pre><code><span><span class="co">## Highest-ranked genes:</span></span>
<span><span class="co">## Bka,RpS2,snoRNA:Me18S-A1576,snoRNA:Me18S-G1620,snoRNA:U14:30Ea,snoRNA:U14:30Eb,Uhg2, MICU1, oaf,Slh, CG15880,Pex12,Saf6, CG17544,Pax, CG17490,lncRNA:CR46268,RpL5,snoRNA:Psi28S-2996, haf,Rab3GAP1, CG11927,mxt, CG15425,lncRNA:CR45294,lncRNA:CR45295,lncRNA:CR45296, AIF,TBCD</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## 173 loci enriched in L5 neurons</span></span></code></pre>
<pre><code><span><span class="co">## Highest-ranked genes:</span></span>
<span><span class="co">## beat-Ic, Ance-3, CG11634, CG13946, Ugt36F1, Ccdc85, dmGlut,lncRNA:CR44587, CG10237, lncRNA:CR43314, eya</span></span></code></pre>
<p>Please note the required <code>cond</code> parameter to this
function, which specifies the filename text that distinguished the two
separate conditions to test. <code>cond[1]</code> must be a text string
present in all, and only all, of the filenames of the
<strong>first</strong> condition to test; <code>cond[2]</code> must be a
text string present in all, and only all, of the filenames of the
<strong>second</strong> condition to test.</p>
<p>The use of <code>cond_names</code> here is optional, and provides
display names for the conditions for downstream visualisation.</p>
<p>Note that when run, <code>differential_binding</code> (or
<code>differential_accessibility</code>) will list the replicates found
under each condition. The package attempts to ensure that condition
search overlaps are avoided, but please check that all replicates have
been correctly assigned.</p>
<p>The functions’ output provides a quick “top ten” list of genes
associated with the most significant, differentially bound loci. This is
provided for a quick verification that the analysis has worked
correctly, and does not imply any special value to the listed genes
beyond that.</p>
<p>Both <code><a href="../reference/differential_binding.html">differential_binding()</a></code> and
<code><a href="../reference/differential_accessibility.html">differential_accessibility()</a></code> return a
<code>DamIDResults</code> S4 object, that forms the basis of downstream
analysis. Printing the object will provide a simple summary of the
analysis results.</p>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># See a summary of the results object</span></span>
<span><span class="va">diff.bsh</span></span></code></pre></div>
<pre><code><span><span class="co">## An object of class 'DamIDResults'</span></span>
<span><span class="co">## Differentially bound regions</span></span>
<span><span class="co">## Comparison: 'L4 neurons' vs 'L5 neurons'</span></span>
<span><span class="co">## - 284 regions enriched in L4 neurons</span></span>
<span><span class="co">## - 173 regions enriched in L5 neurons</span></span>
<span><span class="co">## - 1196 total regions tested</span></span></code></pre>
<p>To fully explore the differentially bound loci, use one of the
visualisation functions below.</p>
</div>
<div class="section level3">
<h3 id="analysing-differential-expression">Analysing differential expression<a class="anchor" aria-label="anchor" href="#analysing-differential-expression"></a>
</h3>
<p>The sample dataset provided is a transcription factor binding
dataset, with pre-called peaks. If analysing gene expression datasets
via RNA Polymerase DamID and <code><a href="../reference/load_data_genes.html">load_data_genes()</a></code>, the
analysis pipeline is similar, with some exceptions:</p>
<ol style="list-style-type: decimal">
<li>No peaks are required. <em><a href="https://bioconductor.org/packages/3.20/damidBind" class="external-link">damidBind</a></em>
calculates signal occupancy over genes directly, and all downstream
analyses are conducted on gene loci.</li>
<li>Optional FDR calling via
<code>load_data_genes(calculate_fdr = TRUE)</code>. <em><a href="https://bioconductor.org/packages/3.20/damidBind" class="external-link">damidBind</a></em>
can determine the FDR for enriched occupancy of RNA Polymerase over gene
bodies, which we have previously shown to be a proxy for gene expression
<span class="citation">(Southall <em>et al</em>, 2013; Marshall <em>et
al</em>, 2016)</span>.</li>
<li>FDR values are not used for DEG analysis, but when present these can
and should be used to filter the Venn and volcano plot outputs via the
<code>fdr_filter_threshold</code> parameter, so that only genes deemed
to be expressed are shown.</li>
<li>The accessor method <code><a href="../reference/expressed.html">expressed()</a></code> provides a quick means
to obtain the list of genes passing an FDR threshold for one of the
analysed conditions.</li>
</ol>
</div>
<div class="section level3">
<h3 id="visualising-data">Visualising data<a class="anchor" aria-label="anchor" href="#visualising-data"></a>
</h3>
<p>All downstream visualisation / data exploration functions take the
<code>DamIDResults</code> object returned from the analysis functions
<code><a href="../reference/differential_binding.html">differential_binding()</a></code> or
<code><a href="../reference/differential_accessibility.html">differential_accessibility()</a></code>.</p>
<div class="section level4">
<h4 id="venn-diagrams-of-differentially-bound-loci">Venn diagrams of differentially bound loci<a class="anchor" aria-label="anchor" href="#venn-diagrams-of-differentially-bound-loci"></a>
</h4>
<p>Venn diagrams are a simple means of visualising the proportion of
loci that are differentially bound between the two tested conditions. To
provide this, <em><a href="https://bioconductor.org/packages/3.20/damidBind" class="external-link">damidBind</a></em>
uses <em><a href="https://CRAN.R-project.org/package=BioVenn" class="external-link">BioVenn</a></em> to
generate proportional Venn diagrams of the two conditions. The set union
represents significant binding peaks that fail to show significant
differences in occupancy; the exclusive regions of each set represent
regions with enriched differential binding in that condition.</p>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_venn.html">plot_venn</a></span><span class="op">(</span><span class="va">diff.bsh</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="damidBind_vignette_files/figure-html/venn-1.png" alt="A venn diagram of significantly bound loci by Bsh in L4 and L5 neurons.  The exclusive parts of each set represent regions that are differentially bound between the two conditions." width="700"><p class="caption">
A venn diagram of significantly bound loci by Bsh in L4 and L5 neurons.
The exclusive parts of each set represent regions that are
differentially bound between the two conditions.
</p>
</div>
</div>
<div class="section level4">
<h4 id="volcano-plots">Volcano plots<a class="anchor" aria-label="anchor" href="#volcano-plots"></a>
</h4>
<p><em><a href="https://bioconductor.org/packages/3.20/damidBind" class="external-link">damidBind</a></em>
comes with a comprehensive volcano plot function. The following gives
some idea as to the capabilities.</p>
<div class="section level5">
<h5 id="the-default-volcano-plot">The default volcano plot<a class="anchor" aria-label="anchor" href="#the-default-volcano-plot"></a>
</h5>
<p>The simple volcano plot already gives a clear picture of
differentially bound loci and their associated genes.</p>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_volcano.html">plot_volcano</a></span><span class="op">(</span></span>
<span>    <span class="va">diff.bsh</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##   Label sampling: coordinates have been centred and scaled.</span></span></code></pre>
<div class="figure">
<img src="damidBind_vignette_files/figure-html/volcanoPlotSimple-1.png" alt="Differential binding of Bsh in L4 and L5 neuronal subtypes.  Genes associated with differentially bound peaks are displayed; the limitations of label overlaps means that only outliers are labelled.  (Dataset from chromosome 2L only)" width="700"><p class="caption">
Differential binding of Bsh in L4 and L5 neuronal subtypes. Genes
associated with differentially bound peaks are displayed; the
limitations of label overlaps means that only outliers are labelled.
(Dataset from chromosome 2L only)
</p>
</div>
</div>
<div class="section level5">
<h5 id="cleaning-up-the-gene-names">Cleaning up the gene names<a class="anchor" aria-label="anchor" href="#cleaning-up-the-gene-names"></a>
</h5>
<p>However, a lot of plot label space can be taken up by generally
uninformative snoRNA / tRNA genes. We can optionally remove these from
the plot using the <code>clean_names=TRUE</code> parameter, to show more
potentially informative labels.</p>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_volcano.html">plot_volcano</a></span><span class="op">(</span></span>
<span>    <span class="va">diff.bsh</span>,</span>
<span>    label_config <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>clean_names <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##   Label sampling: coordinates have been centred and scaled.</span></span></code></pre>
<div class="figure">
<img src="damidBind_vignette_files/figure-html/volcanoPlotCleanNames-1.png" alt="Differential binding of Bsh in L4 and L5 neuronal subtypes.  Genes associated with differentially bound peaks are displayed, after some common, but less useful, gene label classes are removed.  (Dataset from chromosome 2L only)" width="700"><p class="caption">
Differential binding of Bsh in L4 and L5 neuronal subtypes. Genes
associated with differentially bound peaks are displayed, after some
common, but less useful, gene label classes are removed. (Dataset from
chromosome 2L only)
</p>
</div>
<p>Already, that’s becoming a lot clearer.</p>
<p>Note that the relative absence of labels on the L4 side of the plot
does not reflect a paucity of loci, but rather the density of loci on
the plot that prevents labels being drawn without overlaps. To increase
the allowed label overlaps, set the <code>max_overlaps</code> parameter
in the <code>label_config</code> parameter list.</p>
</div>
<div class="section level5">
<h5 id="highlighting-gene-groups">Highlighting gene groups<a class="anchor" aria-label="anchor" href="#highlighting-gene-groups"></a>
</h5>
<p>But, what if we now wanted to highlight genes bound by Bsh, that are
only expressed in L4 neurons? (data from scRNA-seq data, provided in
Supplementary Files 2 &amp; 3 of <span class="citation">(Xu <em>et
al</em>, 2024)</span>). This is straightforward to do:</p>
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">L4_only_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Mp"</span>, <span class="st">"tnc"</span>, <span class="st">"grn"</span>, <span class="st">"rut"</span>, <span class="st">"mtd"</span>, <span class="st">"rdgB"</span>, <span class="st">"Octbeta2R"</span>, <span class="st">"msi"</span>, <span class="st">"Octbeta3R"</span>, <span class="st">"beat-IIIb"</span>, <span class="st">"ap"</span>, <span class="st">"Fili"</span>, <span class="st">"LRP1"</span>, <span class="st">"CG7378"</span>, <span class="st">"CG13698"</span>, <span class="st">"twit"</span>, <span class="st">"CG9336"</span>, <span class="st">"tok"</span>, <span class="st">"CG12991"</span>, <span class="st">"dpr1"</span>, <span class="st">"CG42339"</span>, <span class="st">"beat-IIb"</span>, <span class="st">"mav"</span>, <span class="st">"CG34377"</span>, <span class="st">"alpha-Man-IIb"</span>, <span class="st">"Pli"</span>, <span class="st">"CG32428"</span>, <span class="st">"osp"</span>, <span class="st">"Pka-R2"</span>, <span class="st">"CG15202"</span>, <span class="st">"CG8916"</span>, <span class="st">"CG15894"</span>, <span class="st">"side"</span>, <span class="st">"CG42258"</span>, <span class="st">"CHES-1-like"</span>, <span class="st">"SP2353"</span>, <span class="st">"CG44838"</span>, <span class="st">"Atg1"</span>, <span class="st">"Traf4"</span>, <span class="st">"DIP-beta"</span>, <span class="st">"KCNQ"</span>, <span class="st">"metro"</span>, <span class="st">"nAChRalpha1"</span>, <span class="st">"path"</span>, <span class="st">"CG10527"</span>, <span class="st">"Pde8"</span>, <span class="st">"CG30116"</span>, <span class="st">"CG7985"</span>, <span class="st">"CG1688"</span>, <span class="st">"dpr12"</span>, <span class="st">"pigs"</span>, <span class="st">"Eip63F-1"</span>, <span class="st">"CG14795"</span>, <span class="st">"2mit"</span>, <span class="st">"CG42340"</span>, <span class="st">"BicD"</span>, <span class="st">"CG18265"</span>, <span class="st">"hppy"</span>, <span class="st">"5-HT1A"</span>, <span class="st">"Chd64"</span>, <span class="st">"CG33090"</span>, <span class="st">"Dyb"</span>, <span class="st">"Btk29A"</span>, <span class="st">"Apc"</span>, <span class="st">"Rox8"</span>, <span class="st">"nAChRalpha5"</span>, <span class="st">"CG42748"</span>, <span class="st">"CG3257"</span>, <span class="st">"CG2269"</span>, <span class="st">"beat-IV"</span>, <span class="st">"CG8086"</span>, <span class="st">"glec"</span>, <span class="st">"CG31688"</span>, <span class="st">"oaf"</span>, <span class="st">"Drl-2"</span>, <span class="st">"CG8188"</span>, <span class="st">"aos"</span>, <span class="st">"CG31676"</span>, <span class="st">"REPTOR"</span>, <span class="st">"RabX4"</span>, <span class="st">"alt"</span>, <span class="st">"Pura"</span>, <span class="st">"DIP1"</span>, <span class="st">"ewg"</span>, <span class="st">"side-VIII"</span>, <span class="st">"nAChRalpha7"</span>, <span class="st">"Alh"</span>, <span class="st">"kug"</span>, <span class="st">"Ca-Ma2d"</span>, <span class="st">"bru2"</span>, <span class="st">"CG43737"</span>, <span class="st">"lncRNA:CR44024"</span>, <span class="st">"lncRNA:CR46006"</span>, <span class="st">"Had1"</span>, <span class="st">"CG3961"</span>, <span class="st">"comm"</span>, <span class="st">"Toll-6"</span>, <span class="st">"CG13685"</span>, <span class="st">"tow"</span>, <span class="st">"CG10019"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/plot_volcano.html">plot_volcano</a></span><span class="op">(</span></span>
<span>    <span class="va">diff.bsh</span>,</span>
<span>    label_config <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>    highlight <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>        <span class="st">"L4 specific"</span> <span class="op">=</span> <span class="va">L4_only_genes</span></span>
<span>    <span class="op">)</span>,</span>
<span>    highlight_config <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>        size <span class="op">=</span> <span class="fl">2</span></span>
<span>    <span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="damidBind_vignette_files/figure-html/volcanoPlotHighlightedGenes-1.png" alt="Differential binding of Bsh in L4 and L5 neuronal subtypes.  Genes that are specifically expressed in L4 neurons are highlighted.  (Dataset from chromosome 2L only)" width="700"><p class="caption">
Differential binding of Bsh in L4 and L5 neuronal subtypes. Genes that
are specifically expressed in L4 neurons are highlighted. (Dataset from
chromosome 2L only)
</p>
</div>
<p>Note that the legend guide masks some labels, when positioned inside
the plot on the right-hand side (the default). <code>damidBind</code>
comes with a simple shortcut to display the legend on the left side of
the plot instead, as shown below. It is also possible to place the
legend outside the plot, or to customise the position of the legend
within the plot precisely.</p>
<div class="sourceCode" id="cb57"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_volcano.html">plot_volcano</a></span><span class="op">(</span></span>
<span>    <span class="va">diff.bsh</span>,</span>
<span>    label_config <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>    highlight <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>        <span class="st">"L4 specific"</span> <span class="op">=</span> <span class="va">L4_only_genes</span></span>
<span>    <span class="op">)</span>,</span>
<span>    highlight_config <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>        size <span class="op">=</span> <span class="fl">2</span>,</span>
<span>        legend_inside_pos <span class="op">=</span> <span class="st">'l'</span></span>
<span>    <span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="damidBind_vignette_files/figure-html/volcanoPlotHighlightedGenesLHS-1.png" alt="Differential binding of Bsh in L4 and L5 neuronal subtypes.  Genes that are specifically expressed in L4 neurons are highlighted.  (Dataset from chromosome 2L only).  Legend is now displayed on the LHS within the plot." width="700"><p class="caption">
Differential binding of Bsh in L4 and L5 neuronal subtypes. Genes that
are specifically expressed in L4 neurons are highlighted. (Dataset from
chromosome 2L only). Legend is now displayed on the LHS within the plot.
</p>
</div>
<p>Although this is just a truncated sample from chr 2L, the link
between a subset of Bsh binding and potential upregulation in this
lineage is apparent. (Why do some gene loci label multiple peaks? This
is because a gene may have more than one discrete binding peak
associated with it.)</p>
<p>We could also compare this against genes bound by Bsh and only
expressed in L5 neurons. For clarity, we’ll colour the text labels to
match the conditions, and we’ll reduce the text luminosity to make the
labels easier to read.</p>
<div class="sourceCode" id="cb58"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">L5_only_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Ptth"</span>, <span class="st">"Nep2"</span>, <span class="st">"kek1"</span>, <span class="st">"CG4168"</span>, <span class="st">"kek3"</span>, <span class="st">"CG6959"</span>, <span class="st">"Dtg"</span>, <span class="st">"ND-23"</span>, <span class="st">"Scp2"</span>, <span class="st">"Octalpha2R"</span>, <span class="st">"Hs6st"</span>, <span class="st">"CG16791"</span>, <span class="st">"SKIP"</span>, <span class="st">"LpR1"</span>, <span class="st">"RpL34a"</span>, <span class="st">"Ald1"</span>, <span class="st">"CG10011"</span>, <span class="st">"heph"</span>, <span class="st">"nolo"</span>, <span class="st">"Act42A"</span>, <span class="st">"Fkbp12"</span>, <span class="st">"Pkc53E"</span>, <span class="st">"AstC-R1"</span>, <span class="st">"Muc14A"</span>, <span class="st">"CG33543"</span>, <span class="st">"ChAT"</span>, <span class="st">"Act5C"</span>, <span class="st">"Ptpmeg2"</span>, <span class="st">"fabp"</span>, <span class="st">"CG31221"</span>, <span class="st">"Octbeta1R"</span>, <span class="st">"CG14669"</span>, <span class="st">"sdk"</span>, <span class="st">"Shawl"</span>, <span class="st">"side-V"</span>, <span class="st">"NaCP60E"</span>, <span class="st">"sif"</span>, <span class="st">"OtopLc"</span>, <span class="st">"side-II"</span>, <span class="st">"kuz"</span>, <span class="st">"CG42540"</span>, <span class="st">"Dscam3"</span>, <span class="st">"haf"</span>, <span class="st">"CG42673"</span>, <span class="st">"pdm3"</span>, <span class="st">"tinc"</span>, <span class="st">"CG42750"</span>, <span class="st">"sdt"</span>, <span class="st">"Nuak1"</span>, <span class="st">"Hk"</span>, <span class="st">"scrib"</span>, <span class="st">"tsr"</span>, <span class="st">"dpr20"</span>, <span class="st">"GluRIB"</span>, <span class="st">"CG43902"</span>, <span class="st">"CG44242"</span>, <span class="st">"Dscam2"</span>, <span class="st">"CG44422"</span>, <span class="st">"lncRNA:CR45312"</span>, <span class="st">"Scsalpha1"</span>, <span class="st">"Rop"</span>, <span class="st">"Con"</span>, <span class="st">"Hsc70-3"</span>, <span class="st">"dpr8"</span>, <span class="st">"eag"</span>, <span class="st">"ND-18"</span>, <span class="st">"Nrt"</span>, <span class="st">"CG17839"</span>, <span class="st">"fz"</span>, <span class="st">"CG32137"</span>, <span class="st">"Rh7"</span>, <span class="st">"Sod1"</span>, <span class="st">"CG32052"</span>, <span class="st">"dpr6"</span>, <span class="st">"Hsp67Ba"</span>, <span class="st">"axed"</span>, <span class="st">"GluRIA"</span>, <span class="st">"robo2"</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="fu"><a href="../reference/plot_volcano.html">plot_volcano</a></span><span class="op">(</span></span>
<span>    <span class="va">diff.bsh</span>,</span>
<span>    label_config <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>    highlight <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>        <span class="st">"L4 specific"</span> <span class="op">=</span> <span class="va">L4_only_genes</span>,</span>
<span>        <span class="st">"L5 specific"</span> <span class="op">=</span> <span class="va">L5_only_genes</span></span>
<span>    <span class="op">)</span>,</span>
<span>    highlight_config <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>        size <span class="op">=</span> <span class="fl">2</span>,</span>
<span>        legend_inside_pos <span class="op">=</span> <span class="st">'l'</span>,</span>
<span>        text_col <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>        text_luminosity <span class="op">=</span> <span class="fl">40</span></span>
<span>    <span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="damidBind_vignette_files/figure-html/volcannoPlotMultipleHighlights-1.png" alt="Differential binding of Bsh in L4 and L5 neuronal subtypes.  Genes that are specifically expressed in each subtype are highlighted.   (Dataset from chromosome 2L only)" width="700"><p class="caption">
Differential binding of Bsh in L4 and L5 neuronal subtypes. Genes that
are specifically expressed in each subtype are highlighted. (Dataset
from chromosome 2L only)
</p>
</div>
<p>Here, the L5-specific Bsh peaks show no clear link with upregulated
or downregulated genes between the lineages.</p>
</div>
</div>
<div class="section level4">
<h4 id="browsing-differentially-bound-regions-with-an-igvshiny-browser-window">Browsing differentially-bound regions with an igvShiny browser
window<a class="anchor" aria-label="anchor" href="#browsing-differentially-bound-regions-with-an-igvshiny-browser-window"></a>
</h4>
<p>A volcano plot, while useful, still does not put the binding data
into a genomic context. For this <em><a href="https://bioconductor.org/packages/3.20/damidBind" class="external-link">damidBind</a></em>
provides an interactive Shiny browser window (via <em><a href="https://CRAN.R-project.org/package=Shiny" class="external-link">Shiny</a></em> and
<em><a href="https://bioconductor.org/packages/3.20/igvShiny" class="external-link">igvShiny</a></em>),
where an interactive table of enriched loci allows quick exploration of
these in the genome browser.</p>
<div class="sourceCode" id="cb59"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Interactive, blocking session, uncomment to run</span></span>
<span><span class="co"># browse_igv_regions(diff.bsh)</span></span></code></pre></div>
<p>An example of the interface is shown below:</p>
<div class="float">
<img src="figures/damidBind_L5.png" alt="The interactive IGV browser window of damidBind"><div class="figcaption">The interactive IGV browser window of
damidBind</div>
</div>
<p>The table on the left side of the window can be sorted by clicking on
the column headers, and clicking on any entry will move the IGV browser
window to that locus (+/- a sensible buffer region). All sample reps are
shown, together with the unified binding peaks and a simple pair of
tracks that show the loci that are enriched in both conditions.</p>
</div>
<div class="section level4">
<h4 id="gene-ontology-go-enrichment-plots">Gene Ontology (GO) enrichment plots<a class="anchor" aria-label="anchor" href="#gene-ontology-go-enrichment-plots"></a>
</h4>
<p>Gene ontology is a powerful means of understanding which biological
processes may be changing between two sets of data. Using <em><a href="https://bioconductor.org/packages/3.20/clusterProfiler" class="external-link">clusterProfiler</a></em>
as the backend, <em><a href="https://bioconductor.org/packages/3.20/damidBind" class="external-link">damidBind</a></em>
provides the function <code><a href="../reference/analyse_go_enrichment.html">analyse_go_enrichment()</a></code> to explore
enrichment of genes associated with bound peaks, accessible regions, or
differential expression. In all cases, the underlying gene IDs, rather
than gene names, are used for enrichment analysis.</p>
<p>The example dataset here is minimal, so there is not much to
highlight when we look at just the genes enriched in close proximity to
L5-specific binding peaks.</p>
<div class="sourceCode" id="cb60"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">go.bsh_l4</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/analyse_go_enrichment.html">analyse_go_enrichment</a></span><span class="op">(</span></span>
<span>    <span class="va">diff.bsh</span>,</span>
<span>    direction <span class="op">=</span> <span class="st">"L5"</span>,</span>
<span>    org_db <span class="op">=</span> <span class="fu">org.Dm.eg.db</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/org.Dm.eg.db/man/org.Dm.egBASE.html" class="external-link">org.Dm.eg.db</a></span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span></code></pre>
<div class="figure">
<img src="damidBind_vignette_files/figure-html/gsea-1.png" alt="Enriched GO terms for genes associated with differential Bsh binding in L5 neuron.  Dataset from chromosome 2L only." width="700"><p class="caption">
Enriched GO terms for genes associated with differential Bsh binding in
L5 neuron. Dataset from chromosome 2L only.
</p>
</div>
<p>The return value of this function includes the full analysis table,
including the names of all enriched loci within an ontology term.</p>
</div>
</div>
<div class="section level3">
<h3 id="accessor-methods-and-further-analysis">Accessor methods and further analysis<a class="anchor" aria-label="anchor" href="#accessor-methods-and-further-analysis"></a>
</h3>
<p>The <code>DamIDResults</code> object returned by
<code><a href="../reference/differential_binding.html">differential_binding()</a></code> and
<code><a href="../reference/differential_accessibility.html">differential_accessibility()</a></code> provides accessor methods for
accessing data for further analysis.</p>
<p>The following accessor functions are available for a
<code>DamIDResults</code> object.</p>
<ul>
<li>
<code>analysisTable(object)</code>: Returns the full differential
analysis table (a <code>data.frame</code>).</li>
<li>
<code>enrichedCond1(object)</code>: Returns a
<code>data.frame</code> of regions significantly enriched in the first
condition.</li>
<li>
<code>enrichedCond2(object)</code>: Returns a
<code>data.frame</code> of regions significantly enriched in the second
condition.</li>
<li>
<code>conditionNames(object)</code>: Returns a named character
vector mapping display names to internal identifiers.</li>
<li>
<code>inputData(object)</code>: Returns a list containing the
original input data used for the analysis.</li>
<li>
<code>expressed(object, condition, fdr = 0.05, which = "any")</code>:
Returns a <code>data.frame</code> of genes considered expressed in
‘condition’, based on an FDR threshold of significantly enriched
occupancy. Only available for analyses with FDR calculations, generated
via <code>load_data_genes(calculate_fdr = TRUE)</code>.</li>
</ul>
</div>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Aughey2018" class="csl-entry">
Aughey GN, Estacio Gomez A, Thomson J, Yin H &amp; Southall TD (2018) <a href="https://doi.org/10.7554/elife.32341" class="external-link"><span>CATaDa</span> reveals
global remodelling of chromatin accessibility during stem cell
differentiation in vivo</a>. <em>eLife</em> 7: 1–22
</div>
<div id="ref-Marshall2015" class="csl-entry">
Marshall OJ &amp; Brand AH (2015) <a href="https://doi.org/10.1093/bioinformatics/btv386" class="external-link">Damidseq_pipeline:
An automated pipeline for processing <span>DamID</span> sequencing
datasets.</a> <em>Bioinformatics</em> 31: 3371–3
</div>
<div id="ref-Marshall2017" class="csl-entry">
Marshall OJ &amp; Brand AH (2017) <a href="https://doi.org/10.1038/s41467-017-02385-4" class="external-link">Chromatin state
changes during neural development revealed by in vivo cell-type specific
profiling</a>. <em>Nat Commun</em> 8: 2271
</div>
<div id="ref-Marshall2016" class="csl-entry">
Marshall OJ, Southall TD, Cheetham SW &amp; Brand AH (2016) <a href="https://doi.org/10.1038/nprot.2016.084" class="external-link">Cell-type-specific
profiling of protein–<span>DNA</span> interactions without cell
isolation using targeted <span>DamID</span> with next-generation
sequencing</a>. <em>Nat Protoc</em> 11: 1586–1598
</div>
<div id="ref-Southall2013" class="csl-entry">
Southall TD, Gold KS, Egger B, Davidson CM, Caygill EE, Marshall OJ
&amp; Brand AH (2013) <a href="https://doi.org/10.1016/j.devcel.2013.05.020" class="external-link">Cell-<span>Type-Specific
Profiling</span> of <span>Gene Expression</span> and <span>Chromatin
Binding</span> without <span>Cell Isolation</span>: <span>Assaying RNA
Pol II Occupancy</span> in <span>Neural Stem Cells</span>.</a> <em>Dev
Cell</em> 26: 101–12
</div>
<div id="ref-Steensel2001" class="csl-entry">
van Steensel B, Delrow J &amp; Henikoff S (2001) <a href="https://doi.org/10.1038/85871" class="external-link">Chromatin profiling using targeted
<span>DNA</span> adenine methyltransferase.</a> <em>Nat Genet</em> 27:
304–8
</div>
<div id="ref-Steensel2000" class="csl-entry">
van Steensel B &amp; Henikoff S (2000) <a href="https://doi.org/10.1038/74487" class="external-link">Identification of in vivo
<span>DNA</span> targets of chromatin proteins using tethered dam
methyltransferase.</a> <em>Nat Biotechnol</em> 18: 424–8
</div>
<div id="ref-Xu2024" class="csl-entry">
Xu C, Ramos TB, Marshall OJ &amp; Doe CQ (2024) <a href="https://doi.org/10.7554/eLife.90136" class="external-link">Notch signaling and
<span>Bsh</span> homeodomain activity are integrated to diversify
<span>Drosophila</span> lamina neuron types</a>. <em>eLife</em> 12:
RP90136
</div>
</div>
</div>
<div class="section level2">
<h2 id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<pre><code><span><span class="co">## R version 4.4.0 (2024-04-24)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">## Running under: Ubuntu 22.04.5 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_AU.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_AU.UTF-8        LC_COLLATE=en_AU.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_AU.UTF-8    LC_MESSAGES=en_AU.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_AU.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_AU.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: Etc/UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">## [8] base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] ensembldb_2.30.0        AnnotationFilter_1.30.0 GenomicFeatures_1.58.0 </span></span>
<span><span class="co">##  [4] AnnotationDbi_1.68.0    Biobase_2.66.0          GenomicRanges_1.58.0   </span></span>
<span><span class="co">##  [7] GenomeInfoDb_1.42.3     IRanges_2.40.1          S4Vectors_0.44.0       </span></span>
<span><span class="co">## [10] BiocGenerics_0.52.0     damidBind_0.99.4        DT_0.34.0              </span></span>
<span><span class="co">## [13] BiocStyle_2.34.0       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] later_1.4.4                 splines_4.4.0              </span></span>
<span><span class="co">##   [3] BiocIO_1.16.0               bitops_1.0-9               </span></span>
<span><span class="co">##   [5] ggplotify_0.1.3             filelock_1.0.3             </span></span>
<span><span class="co">##   [7] tibble_3.3.0                R.oo_1.27.1                </span></span>
<span><span class="co">##   [9] XML_3.99-0.19               lifecycle_1.0.4            </span></span>
<span><span class="co">##  [11] httr2_1.2.1                 org.Dm.eg.db_3.20.0        </span></span>
<span><span class="co">##  [13] lattice_0.22-7              backports_1.5.0            </span></span>
<span><span class="co">##  [15] NOISeq_2.50.0               magrittr_2.0.4             </span></span>
<span><span class="co">##  [17] limma_3.62.2                sass_0.4.10                </span></span>
<span><span class="co">##  [19] rmarkdown_2.30              jquerylib_0.1.4            </span></span>
<span><span class="co">##  [21] yaml_2.3.10                 plotrix_3.8-4              </span></span>
<span><span class="co">##  [23] otel_0.2.0                  httpuv_1.6.16              </span></span>
<span><span class="co">##  [25] ggtangle_0.0.7              cowplot_1.2.0              </span></span>
<span><span class="co">##  [27] DBI_1.2.3                   RColorBrewer_1.1-3         </span></span>
<span><span class="co">##  [29] abind_1.4-8                 zlibbioc_1.52.0            </span></span>
<span><span class="co">##  [31] Rtsne_0.17                  purrr_1.1.0                </span></span>
<span><span class="co">##  [33] R.utils_2.13.0              RCurl_1.98-1.17            </span></span>
<span><span class="co">##  [35] yulab.utils_0.2.1           rappdirs_0.3.3             </span></span>
<span><span class="co">##  [37] GenomeInfoDbData_1.2.13     enrichplot_1.26.6          </span></span>
<span><span class="co">##  [39] ggrepel_0.9.6               tidytree_0.4.6             </span></span>
<span><span class="co">##  [41] pkgdown_2.1.3               svglite_2.2.2              </span></span>
<span><span class="co">##  [43] codetools_0.2-20            DelayedArray_0.32.0        </span></span>
<span><span class="co">##  [45] DOSE_4.0.1                  xml2_1.4.1                 </span></span>
<span><span class="co">##  [47] tidyselect_1.2.1            futile.logger_1.4.3        </span></span>
<span><span class="co">##  [49] aplot_0.2.9                 UCSC.utils_1.2.0           </span></span>
<span><span class="co">##  [51] farver_2.1.2                matrixStats_1.5.0          </span></span>
<span><span class="co">##  [53] BiocFileCache_2.14.0        GenomicAlignments_1.42.0   </span></span>
<span><span class="co">##  [55] jsonlite_2.0.0              randomcoloR_1.1.0.1        </span></span>
<span><span class="co">##  [57] systemfonts_1.3.1           tools_4.4.0                </span></span>
<span><span class="co">##  [59] ggnewscale_0.5.2            progress_1.2.3             </span></span>
<span><span class="co">##  [61] treeio_1.30.0               Rcpp_1.1.0                 </span></span>
<span><span class="co">##  [63] glue_1.8.0                  SparseArray_1.6.2          </span></span>
<span><span class="co">##  [65] xfun_0.54                   qvalue_2.38.0              </span></span>
<span><span class="co">##  [67] MatrixGenerics_1.18.1       dplyr_1.1.4                </span></span>
<span><span class="co">##  [69] withr_3.0.2                 formatR_1.14               </span></span>
<span><span class="co">##  [71] BiocManager_1.30.26         fastmap_1.2.0              </span></span>
<span><span class="co">##  [73] digest_0.6.37               BioVenn_1.1.3              </span></span>
<span><span class="co">##  [75] mime_0.13                   R6_2.6.1                   </span></span>
<span><span class="co">##  [77] gridGraphics_0.5-1          textshaping_1.0.4          </span></span>
<span><span class="co">##  [79] colorspace_2.1-2            GO.db_3.20.0               </span></span>
<span><span class="co">##  [81] biomaRt_2.62.1              RSQLite_2.4.3              </span></span>
<span><span class="co">##  [83] R.methodsS3_1.8.2           tidyr_1.3.1                </span></span>
<span><span class="co">##  [85] generics_0.1.4              data.table_1.17.8          </span></span>
<span><span class="co">##  [87] rtracklayer_1.66.0          prettyunits_1.2.0          </span></span>
<span><span class="co">##  [89] httr_1.4.7                  htmlwidgets_1.6.4          </span></span>
<span><span class="co">##  [91] S4Arrays_1.6.0              pkgconfig_2.0.3            </span></span>
<span><span class="co">##  [93] gtable_0.3.6                blob_1.2.4                 </span></span>
<span><span class="co">##  [95] S7_0.2.0                    XVector_0.46.0             </span></span>
<span><span class="co">##  [97] clusterProfiler_4.14.6      htmltools_0.5.8.1          </span></span>
<span><span class="co">##  [99] bookdown_0.45               fgsea_1.32.4               </span></span>
<span><span class="co">## [101] ProtGenerics_1.38.0         scales_1.4.0               </span></span>
<span><span class="co">## [103] png_0.1-8                   ggfun_0.2.0                </span></span>
<span><span class="co">## [105] lambda.r_1.2.4              knitr_1.50                 </span></span>
<span><span class="co">## [107] rstudioapi_0.17.1           reshape2_1.4.4             </span></span>
<span><span class="co">## [109] rjson_0.2.23                checkmate_2.3.3            </span></span>
<span><span class="co">## [111] nlme_3.1-168                curl_7.0.0                 </span></span>
<span><span class="co">## [113] cachem_1.1.0                stringr_1.5.2              </span></span>
<span><span class="co">## [115] BiocVersion_3.20.0          parallel_4.4.0             </span></span>
<span><span class="co">## [117] restfulr_0.0.16             desc_1.4.3                 </span></span>
<span><span class="co">## [119] pillar_1.11.1               grid_4.4.0                 </span></span>
<span><span class="co">## [121] vctrs_0.6.5                 promises_1.5.0             </span></span>
<span><span class="co">## [123] dbplyr_2.5.1                cluster_2.1.8.1            </span></span>
<span><span class="co">## [125] xtable_1.8-4                evaluate_1.0.5             </span></span>
<span><span class="co">## [127] futile.options_1.0.1        cli_3.6.5                  </span></span>
<span><span class="co">## [129] compiler_4.4.0              Rsamtools_2.22.0           </span></span>
<span><span class="co">## [131] rlang_1.1.6                 crayon_1.5.3               </span></span>
<span><span class="co">## [133] igvShiny_1.2.0              labeling_0.4.3             </span></span>
<span><span class="co">## [135] plyr_1.8.9                  forcats_1.0.1              </span></span>
<span><span class="co">## [137] fs_1.6.6                    stringi_1.8.7              </span></span>
<span><span class="co">## [139] BiocParallel_1.40.2         Biostrings_2.74.1          </span></span>
<span><span class="co">## [141] lazyeval_0.2.2              V8_8.0.1                   </span></span>
<span><span class="co">## [143] GOSemSim_2.32.0             Matrix_1.7-4               </span></span>
<span><span class="co">## [145] RcppHNSW_0.6.0              hms_1.1.4                  </span></span>
<span><span class="co">## [147] patchwork_1.3.2             bit64_4.6.0-1              </span></span>
<span><span class="co">## [149] ggplot2_4.0.0               KEGGREST_1.46.0            </span></span>
<span><span class="co">## [151] statmod_1.5.1               shiny_1.11.1               </span></span>
<span><span class="co">## [153] SummarizedExperiment_1.36.0 AnnotationHub_3.14.0       </span></span>
<span><span class="co">## [155] igraph_2.2.1                memoise_2.0.1              </span></span>
<span><span class="co">## [157] bslib_0.9.0                 ggtree_3.12.0              </span></span>
<span><span class="co">## [159] fastmatch_1.1-6             bit_4.6.0                  </span></span>
<span><span class="co">## [161] ape_5.8-1                   gson_0.1.0</span></span></code></pre>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Owen Marshall.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
